### PR TITLE
`CI`: don't make releases until `release-checks` pass

### PR DESCRIPTION
**Pull request by NachoSoto**

### CODE REVIEW

**Comment by NachoSoto** [comment]
@vegaro am I missing something or it would be a good idea to make this part of `hold` too?

**Comment by NachoSoto** [comment]
That job checks that there's no tag or release created yet. If we approve `hold` before that finishes running it might fail.

**Review by vegaro — APPROVED** [review]

File: .circleci/config.yml [hunk 1]
```diff
@@ -728,9 +728,6 @@ workflows:
           name: xcode-13.2.1
           xcode_version: '13.2.1'
           <<: *release-branches-and-main
```
[the remark this answers is not in the document]
Comment (vegaro) [reply .circleci/config.yml:730]: Actually... These run on `main` too, so I think we are fine... Unless something changes in the release branch

[the remark this answers is not in the document]
Comment (NachoSoto) [reply .circleci/config.yml:730]: Sounds good. Plus I think that won't work if these jobs aren't part of `deploy` (that's why I moved `release-checks` down).

**Pull request merged**